sys/freebsd/socket_inet_tcp.txt

```diff
@@ -82,7 +82,10 @@ tcp_fastopen {
 
 getsockopt$inet_tcp_TCP_RXTLS_MODE(fd sock_tcp, level const[IPPROTO_TCP], optname const[TCP_RXTLS_MODE], optval ptr[out, int32], optlen ptr[inout, len[optval, int32]])
 getsockopt$inet_tcp_TCP_TXTLS_MODE(fd sock_tcp, level const[IPPROTO_TCP], optname const[TCP_TXTLS_MODE], optval ptr[out, int32], optlen ptr[inout, len[optval, int32]])
-setsockopt$inet_tcp_TCP_TXTLS_MODE(fd sock_tcp, level const[IPPROTO_TCP], optname const[TCP_TXTLS_MODE], optval ptr[in, int32], optlen len[optval])
+setsockopt$inet_tcp_TCP_TXTLS_MODE(fd sock_tcp, level const[IPPROTO_TCP], optname const[TCP_TXTLS_MODE], optval ptr[in, flags[tls_tx_mode, int32]], optlen len[optval])
+
+tls_tx_mode = TCP_TLS_MODE_NONE, TCP_TLS_MODE_SW, TCP_TLS_MODE_IFNET, TCP_TLS_MODE_TOE
```

Comment:
We end to group flag definitions together, so I'd put these below the syscall definitions.

Reply:
I thought you grouped the ones used in `tls_enable {}` after its definition, which comes after its use in `setsockopt$inet_tcp_TCP_[RT]XTLS_ENABLE`. This seems to be consistent with the rest of the file. Therefore I put the `tls_tx_mode` after its usage in `setsockopt$inet_tcp_TCP_TXTLS_MODE`. But I don't care that much. If you prefer, I'm happy to move the definition `tls_tx_mode`.

Reply:
I was just following a pattern of

<group of related psuedo syscall definitions>
<related data structures and flags>

I see we are generally not as consistent about it as I thought, so I'm ok with it as-is.